src/components/Note/NoteReactionList/NoteReactionBar/NoteReactionBar.test.tsx

```diff
@@ -1,40 +1,57 @@
+import {getEmojiWithSkinTone} from "utils/reactions";
 import {NoteReactionBar} from "./NoteReactionBar";
 
+const renderNoteReactionBar = (close?: () => void, click?: () => void, overwrite?: Partial<ApplicationState>) =>
+  render(
+    <Provider store={getTestStore(overwrite)}>
+      <NoteReactionBar closeReactionBar={close ?? jest.fn()} reactions={[]} handleClickReaction={click ?? jest.fn()} />
+    </Provider>
+  );
+
 describe("NoteReactionBar", () => {
   it("renders the reaction buttons correctly", () => {
-    render(<NoteReactionBar closeReactionBar={() => {}} reactions={[]} handleClickReaction={() => {}} />);
-
+    renderNoteReactionBar();
     REACTION_EMOJI_MAP.forEach((emoji) => {
-      expect(screen.getByText(emoji)).toBeInTheDocument();
+      expect(screen.getByText(emoji.emoji)).toBeInTheDocument();
     });
   });
 
   it("calls handleClickBar when a reaction button is clicked", () => {
-    const handleClickReaction = jest.fn();
-    render(<NoteReactionBar closeReactionBar={() => {}} reactions={[]} handleClickReaction={handleClickReaction} />);
+    const handleClickReactionFunction = jest.fn();
+    renderNoteReactionBar(undefined, handleClickReactionFunction);
     fireEvent.click(screen.getByText("👍"));
-    expect(handleClickReaction).toHaveBeenCalled();
+    expect(handleClickReactionFunction).toHaveBeenCalled();
   });
 
   it("closes the reaction bar when Escape key is pressed", () => {
-    const closeReactionBar = jest.fn();
-    render(<NoteReactionBar closeReactionBar={closeReactionBar} reactions={[]} handleClickReaction={() => {}} />);
+    const closeReactionBarFunction = jest.fn();
+    renderNoteReactionBar(closeReactionBarFunction);
     fireEvent.keyDown(document, {key: "Escape"});
-    expect(closeReactionBar).toHaveBeenCalled();
+    expect(closeReactionBarFunction).toHaveBeenCalled();
   });
 
   it("focuses the first emoji button after TAB is pressed on the last emoji button", async () => {
-    render(<NoteReactionBar closeReactionBar={() => {}} reactions={[]} handleClickReaction={() => {}} />);
-
-    const firstEmojiButton = screen.getByText([...REACTION_EMOJI_MAP.values()].at(0)!);
-    const lastEmojiButton = screen.getByText([...REACTION_EMOJI_MAP.values()].at(-1)!);
+    renderNoteReactionBar(jest.fn(), jest.fn(), getTestApplicationState());
+    const firstEmojiButton = screen.getByText([...REACTION_EMOJI_MAP.values()].at(0)!.emoji);
+    const lastEmojiButton = screen.getByText([...REACTION_EMOJI_MAP.values()].at(-1)!.emoji);
 
     lastEmojiButton.focus();
     expect(lastEmojiButton).toHaveFocus();
     await userEvent.tab();
     expect(firstEmojiButton).toHaveFocus();
   });
+
+  it("render emoji in different skin tone", () => {
+    renderNoteReactionBar(undefined, undefined, {skinTone: {name: "dark", component: "🏿"}});
+    REACTION_EMOJI_MAP.forEach((emoji) => {
+      expect(screen.getByText(getEmojiWithSkinTone(emoji, {name: "dark", component: "🏿"}))).toBeInTheDocument();
```

Comment:
Is there a way to get the skin tone from the state?

Reply:
`getTestStore()` in the `<Provider/>` returns a store where you could get the state. However, I believe it would be not the best practice to test the store itself. Instead, we should focus about unit testing what the impact of the changed state is (i.e. changed components etc)